CodeModel/src/main/java/org/openzen/zenscript/codemodel/type/member/MemberSet.java

```diff
@@ -58,10 +58,17 @@ public Optional<StaticCallable> findSuffixConstructor(String suffix) {
 	@Override
 	public Optional<InstanceCallableMethod> findCaster(TypeID toType) {
 		MethodID id = MethodID.caster(toType);
-		if (!instanceMethods.containsKey(id))
-			return Optional.empty();
+		if (instanceMethods.containsKey(id)) {
+			return instanceMethods.get(id).stream().findFirst();
+		}
 
-		return instanceMethods.get(id).stream().findFirst();
+		// ToDo: When we have `var a = new List<string>(); var a as string[];`
+		//  Then the MethodID int he instanceMethods map is List<T> but its header returns string?
+		return instanceMethods.keySet().stream()
```

Comment:
❗ Instead of comparing the headers, we probably should add the instantiated Method to this instead of the original header.